src/data/arena-tag.ts

```diff
@@ -988,15 +1019,24 @@ class StickyWebTag extends ArenaTrapTag {
 
   onAdd(arena: Arena, quiet = false): void {
     super.onAdd(arena);
-    const source = this.sourceId ? globalScene.getPokemonById(this.sourceId) : null;
-    if (!quiet && source) {
-      globalScene.phaseManager.queueMessage(
-        i18next.t("arenaTag:stickyWebOnAdd", {
-          moveName: this.getMoveName(),
-          opponentDesc: source.getOpponentDescriptor(),
-        }),
-      );
+
+    // We assume `quiet=true` means "just add the bloody tag no questions asked"
+    if (quiet) {
+      return;
     }
+
+    const source = this.getSourcePokemon();
+    if (!source) {
+      console.warn(`Failed to get source Pokemon for SpikesTag on add message; id: ${this.sourceId}`);
+      return;
+    }
+
+    globalScene.phaseManager.queueMessage(
+      i18next.t("arenaTag:spikesOnAdd", {
```

Comment:
```suggestion
      i18next.t("arenaTag:stickyWebOnAdd", {
```
Some sort of copypaste mistake?

Reply:
Mb lmao